File: backend/app/schemas/blog_entry.py
"""
Blog entry-related Pydantic schemas.
"""
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, Field


class BlogEntryBase(BaseModel):
    """Base schema for blog entry."""
    title: str = Field(..., max_length=500)
    content: str
    excerpt: Optional[str] = None
    status: str = Field(default="draft", pattern="^(draft|published|archived)$")
    is_public: bool = False
    is_featured: bool = False


class BlogEntryCreate(BlogEntryBase):
    """Schema for creating blog entry."""
    pass


class BlogEntryUpdate(BaseModel):
    """Schema for updating blog entry."""
    title: Optional[str] = Field(None, max_length=500)
    content: Optional[str] = None
    excerpt: Optional[str] = None
    status: Optional[str] = Field(None, pattern="^(draft|published|archived)$")
    is_public: Optional[bool] = None
    is_featured: Optional[bool] = None


class BlogEntryResponse(BlogEntryBase):
    """Schema for blog entry response."""
    id: UUID
    user_id: UUID
    slug: Optional[str] = None
    view_count: int
    published_at: Optional[datetime] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/src/types/blogEntry.ts
/**
 * Blog entry related types
 */

export interface BlogEntry {
  id: string;
  user_id: string;
  title: string;
  content: string;
  excerpt: string | null;
  slug: string | null;
  status: 'draft' | 'published' | 'archived';
  is_public: boolean;
  is_featured: boolean;
  view_count: number;
  published_at: string | null;
  created_at: string;
  updated_at: string | null;
}

export interface BlogEntryCreate {
  title: string;
  content: string;
  excerpt?: string;
  status?: 'draft' | 'published' | 'archived';
  is_public?: boolean;
  is_featured?: boolean;
}

export interface BlogEntryUpdate {
  title?: string;
  content?: string;
  excerpt?: string;
  status?: 'draft' | 'published' | 'archived';
  is_public?: boolean;
  is_featured?: boolean;
}

export interface PaginatedBlogEntries {
  items: BlogEntry[];
  total: number;
  page: number;
  limit: number;
  total_pages: number;
}
